src/VisualStudio.Roslyn.SDK/ComponentDebugger/CapabilityProvider.cs

```diff
@@ -0,0 +1,47 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Immutable;
+using System.ComponentModel.Composition;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.VisualStudio.ProjectSystem;
+
+namespace Roslyn.ComponentDebugger
+{
+    [Export(ExportContractNames.Scopes.ConfiguredProject, typeof(IProjectCapabilitiesProvider))]
+    [AppliesTo(ProjectCapabilities.CSharp)]
+    public class CapabilityProvider : ConfiguredProjectCapabilitiesProviderBase
+    {
+        [ImportingConstructor]
+        [System.Obsolete("This exported object must be obtained through the MEF export provider.", error: true)]
+        public CapabilityProvider(ConfiguredProject configuredProject)
+            : base(nameof(CapabilityProvider), configuredProject)
+        {
+        }
+
+        protected override async Task<ImmutableHashSet<string>> GetCapabilitiesAsync(CancellationToken cancellationToken)
+        {
+            // an alternative design could be to have 'IsRoslynComponent' just define the <Capability... directly in the managed.core targets
+            // but that would require a specific roslyn version to work, this allows it to be backwards compatible with older SDKs
+            var caps = Empty.CapabilitiesSet;
+            if (await IsRoslynComponentAsync(this.ConfiguredProject, cancellationToken).ConfigureAwait(false))
```

Comment:
This is mixing dataflow and live project data, which you can't do.

You need to listen to and pull on the ProjectInstance via dataflow, here's an example: https://github.com/dotnet/project-system/blob/b7fd4c439d1d39b656d89a8b88464fabffcc8e59/src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Web/ProjectTypeGuidsDataSource.cs.

Reply:
@davkean Oh, interesting. There are classes in the devdiv repo that use this exact pattern. The base class seems to subscribe to events, so I figured it was designed to explicitly not need dataflow stuff. 

I'll try and use the example to create a dataflow version.